.ci/setup-job/action.yml

```diff
@@ -0,0 +1,55 @@
+name: 'setup job'
```

Comment:
I'll prefer to have one for rust and one for dart ( they can call a common one if we want for env and just).

This will scale a bit better and will allow us to have for example in the rust one a `toolchain` option that by default will use `rust-toolchain.toml` but if specified it will install the passed one. So we don't have a special case for rust fmt.

Regarding the installation of libclang I think we can always doing it when we install dart, it's only 10 seconds and doing this we don't need to know if we are generating code or not.

Reply:
But if we go for for a setup per language then we cannot install the common stuff in each of them or the installation will be duplicate 😒
Not sure what to do then.

Reply:
We don't need to customize the components, we are installing "default" normally and we can do it also in this case, will be a bit more wasted time but not much. 
I think it does not have to be another input but we can reuse the same, instead of true we pass the toolchain version.
Do you think is worth it reconsider if we need to specify only the toolchain?

Reply:
> instead of true we pass the toolchain version.

but then we have to parse the toolchain version or duplicate it which I would want to avoid,
it either involves additional tools, or is prone to getting out of sync.

Through I guess we could pass the toolchain version or true.

I'm still not sure what the problem is with the way it's currently done (besides not using it for fmt), I mean you say
it doesn't scale well, but scale to where?

Reply:
like if add flutter or other languages, this function will bloat a bit. But since is not super easy to reause common part we can keep like it is.

Reply:
> Through I guess we could pass the toolchain version or true.

Yes that was I was trying to say. 
if we can pass `rust: true` or `rust: nightly` should not be too much work on the function side do decide what do.